src/hooks/useAsyncStorage.ts

```diff
@@ -6,10 +7,16 @@ export function useAsyncStorage(key: string) {
     AsyncStorage.getItem(key),
   );
 
-  const setItem = (value: string) =>
-    AsyncStorage.setItem(key, value).catch((error) =>
-      console.error(`[LifeOmicSDK]:${error}`),
-    );
+  const setItem = useCallback(
+    (value: string) => {
+      if (value !== itemResult.data) {
+        AsyncStorage.setItem(key, value).catch((error) =>
+          console.error(`[LifeOmicSDK]:${error}`),
+        );
+      }
+    },
+    [key, itemResult.data],
```

Comment:
Does `itemResult.data` somehow update after one calls `setItem('value')`?  I wouldn't think so, so couldn't something like this happen?

- `itemResult.data` is `1`
- `setItem('2')` -> AsyncStorage is used
- `setItem('2')` -> AsyncStorage is used again
- `setItem('2')` -> AsyncStorage is used again

Reply:
Ahh yeah I'll test this but I bet you are correct.